pkg/render/intrusiondetection/dpi/dpi.go

```diff
@@ -114,6 +124,19 @@ func (d *dpiComponent) Objects() (objsToCreate, objsToDelete []client.Object) {
 			d.dpiClusterRoleBinding(),
 			d.dpiDaemonset(),
 		)
+		// We also want these permissions present for a standalone
+		// cluster that has DPI resource configured
+		if !d.cfg.ManagementCluster && !d.cfg.ManagedCluster {
```

Comment:
Instead of duplicating this logic here and in the above branch, can't we just do this below?

```
if d.cfg.ManagementCluster || !d.cfg.ManagedCluster {
  // Always create permission for DPI to write to Linseed in management and standalone clusters.
} else {
  // Remoe the permissions otherwise. These only need to exist in clusters that are running Linseed.
}
```

Reply:
This means that we will have these roles installed always for standalone regardless whether we have DPI CR configured or not. Also, when removing DPI, these roles will still be present. It does simplify the logic and if it is something we can live with, I have no problem with this.

Reply:
Talked offine and decided on a if else if statement to cover all this.